src/test/java/com/puppycrawl/tools/checkstyle/DefaultLoggerTest.java

```diff
@@ -166,4 +177,78 @@ public void testFinishLocalSetup() {
         assertNotNull(dl, "instance should not be null");
     }
 
+    /**
+     * Verifies that the language specified with the system property {@code user.language} exists.
+     */
+    @Test
+    public void testLanguageIsValid() {
+        final String language = DEFAULT_LOCALE.getLanguage();
+        if (!language.isEmpty()) {
+            assertThat("Invalid language",
+                    Arrays.asList(Locale.getISOLanguages()), hasItem(language));
+        }
+    }
+
+    /**
+     * Verifies that the country specified with the system property {@code user.country} exists.
+     */
+    @Test
+    public void testCountryIsValid() {
+        final String country = DEFAULT_LOCALE.getCountry();
+        if (!country.isEmpty()) {
+            assertThat("Invalid country",
+                    Arrays.asList(Locale.getISOCountries()), hasItem(country));
+        }
+    }
+
+    /**
+     * Verifies that if the language is specified explicitly (and it is not English),
+     * the message does not use the default value.
+     */
+    @Test
+    public void testLocaleIsSupported() {
+        final String language = DEFAULT_LOCALE.getLanguage();
+        if (!language.isEmpty() && !Locale.ENGLISH.getLanguage().equals(language)) {
+            final DefaultLogger.LocalizedMessage message = createSampleLocalizedMessage();
+            assertThat("Unsupported language: " + DEFAULT_LOCALE,
+                    message.getMessage(), not("Error auditing {0}"));
+        }
+    }
+
+    @DefaultLocale("fr")
+    @Test
+    public void testCleatBundleCache() {
```

Comment:
what force you to do `@DefaultLocale("fr")` ?
we do not do this in other places. we have special CI execution for each locale we support.

Reply:
This to check 2 goals:
Primary goal: To check if the BUNDLE_CACHE is being cleared.
Secondary goal: Check if the translation is taking place

Similar test is present in ViolatioTest
https://github.com/checkstyle/checkstyle/blob/6c62d23e3e55fc0326eadab3737d80f1f04d335a/src/test/java/com/puppycrawl/tools/checkstyle/api/ViolationTest.java#L299-L305